docs/development/setup-advanced.md

```diff
@@ -58,51 +58,70 @@ ignoring the parts about `vagrant` (since you're not using it).
 ## Installing directly on Windows 10 with WSL 2
 
 Zulip's development environment is most easily set up on Windows using
-the [WSL 2](https://docs.microsoft.com/en-us/windows/wsl/wsl2-about)
+the Windows Subsystem for Linux [(WSL 2)]
+(https://docs.microsoft.com/en-us/windows/wsl/wsl2-about)
 installation method described here.
 
-1. Install WSL 2 by following the instructions provided by Microsoft
-   [here](https://docs.microsoft.com/en-us/windows/wsl/install).
+1. Enable virtualization through your BIOS settings.
+   This sequence depends on your specific hardware and brand, but here
+   are [some basic instructions.]
+   (https://www.thewindowsclub.com/disable-hardware-virtualization-in-windows-10)
 
-1. Install the `Ubuntu 18.04` Linux distribution from the Microsoft
-   Store.
+1. Install WSL 2 by following the instructions provided by Microsoft
+   [here](https://docs.microsoft.com/en-us/windows/wsl/setup/environment).
 
-1. Launch the `Ubuntu 18.04` shell and run the following commands:
+1. Launch the Ubuntu shell as an administrator and run the following command:
 
    ```bash
    sudo apt update && sudo apt upgrade
+   ```
+
+1. Create a new SSH-2 Key for the WSL-2 Virtual Machine and add it to your
+   github account. [Instructions]
+   (https://docs.github.com/en/authentication/connecting-to-github-with-ssh/adding-a-new-ssh-key-to-your-github-account)
+   A SSH key linked to your current computer will not work within the virtual machine.
+
+1. Install the server with the following command:
+
+   ```bash
    sudo apt install rabbitmq-server memcached redis-server postgresql
    ```
 
 1. Open `/etc/rabbitmq/rabbitmq-env.conf` using e.g.:
 
    ```bash
-   sudo vim /etc/rabbitmq/rabbitmq-env.conf
+   sudo nano /etc/rabbitmq/rabbitmq-env.conf
    ```
 
-   Add the following lines at the end of your file and save:
+   Confirm the following lines are at the end of your file,
+   and add them if not present. Then save and exit.
 
    ```ini
    NODE_IP_ADDRESS=127.0.0.1
    NODE_PORT=5672
    ```
 
-1. Make sure you are inside the WSL disk and not in a Windows mounted disk.
+1. Run the command below to make sure you are inside the WSL disk and not in a Windows mounted disk.
    You will run into permission issues if you run `provision` from `zulip`
    in a Windows mounted disk.
 
    ```bash
    cd ~  # or cd /home/USERNAME
    ```
 
-1. [Clone your fork of the Zulip repository][zulip-rtd-git-cloning]
-   and [connecting the Zulip upstream repository][zulip-rtd-git-connect]:
+1. Create your fork of the Zulip repository at [GitHub](github.com/zulip)
+   The main server app is zulip/zulip in the top left-hand box. [Instructions]
+   (https://docs.github.com/en/get-started/quickstart/fork-a-repo)
```

Comment:
"Top left-hand box" feels too specific -- it could get stale if GitHub rearranges their UI (as they periodically do). We can instead link directly to the repo: github.com/zulip/zulip.

But better yet: this is a step that isn't any different between WSL and any other environment. So we can use the same instructions as elsewhere. Saying something like "Create your fork of the Zulip server repository" with a link to
https://zulip.readthedocs.io/en/latest/git/cloning.html#step-1a-create-your-fork would do it.

Reply:
Thoughts:
Good idea about linking directly to zulip/zulip.

Second - Unless it is better to use the http address, I'll use the [zulip-rtd-<extension>] address, if I can figure it out.  I haven't had luck locating the markup guide for these links, but I think I can deduce it.  This seems to be the standard formatting for the file.  Let me know if you happen to have the documentation handy/know how this works.

Third - What would you think of adding the documentation link while leaving the link to the GitHub instructions?  The GitHub link is the same one found here: https://zulip.readthedocs.io/en/latest/git/cloning.html#step-1a-create-your-fork.  I think this helps to keep focus on the installation instead of having to switch over to the git instructions in the middle.  I find this more focused style to be helpful, but perhaps others don't.

Reply:
RE second thought:  I'll look further through the Sphinx documentation later in the day (or tomorrow), but right now I don't know how to make the appropriate link.  I thought I saw the pattern with it, but it didn't hold as I looked at more links.  I have the one for the cloning documentation, which is just below the fork documentation on the same page, so I'll use it for now and mark a TODO in the file.

Reply:
> Second - Unless it is better to use the http address, I'll use the [zulip-rtd-] address, if I can figure it out. I haven't had luck locating the markup guide for these links, but I think I can deduce it. This seems to be the standard formatting for the file. Let me know if you happen to have the documentation handy/know how this works.

Huh, yeah, that is puzzling! I would have thought that something like this (which is in this file in main):
```
1. [Clone your fork of the Zulip repository][zulip-rtd-git-cloning]
```
meant there was somewhere else in the file a line like this:
```
[zulip-rtd-git-cloning]: https://zulip.readthedocs.io/en/latest/git/cloning.html#step-1b-clone-to-your-machine
```
because that's the standard Markdown way to make a link like that.

But I don't see a line like that. And yet https://zulip.readthedocs.io/en/latest/development/setup-advanced.html shows that it does successfully build and makes a link to https://zulip.readthedocs.io/en/latest/git/cloning.html#step-1b-clone-to-your-machine .

I am curious about what magic is going on there, but it also seems rather too magic. I'd be inclined to make the link in the usual Markdown way -- either with the link inline or explicit on a separate line. In this file, see the mentions of "remote server", "user accounts", and "provision-help" for examples of each style; note that for a link inside the documentation, we use a relative URL.

Reply:
> The GitHub link is the same one found here: https://zulip.readthedocs.io/en/latest/git/cloning.html#step-1a-create-your-fork. I think this helps to keep focus on the installation instead of having to switch over to the git instructions in the middle. I find this more focused style to be helpful, but perhaps others don't.

I think that's fine. In general the risk with that is that there's more detail at the other page, and if it gets copied into several different sets of instructions that use that step then they'll get out of sync over time: we'll make improvements to the explanation at one but the improvements won't make it to the others (because it'd be more work to do so, and/or because the person making the change doesn't notice the others), or something will change so that they need an update but the update won't make it to all the copies. Also, for readers who do know what a given step means, a long explanation makes it harder to read than a short sentence that just names what to do (and provides key details like a link to the repository in question).

But here, the instructions for this step at the other page aren't much more than a link to that page of GitHub docs. So those risks don't really apply.

Reply:
I will switch this to a relative path.  Thank you for the explanation.  It seems it is magic to everyone, because no one has answered in the chat, either. 
I agree about the instructions to fork the repo and the ideas about keeping instructions up to date in general.  This is already committed with both sets of links.  The other zulip docs link needs to be updated, which I will do shortly.

Reply:
Fixed in commit 0f768cf.

Reply:
> It seems it is magic to everyone, because no one has answered in the chat, either.

Yeah, it may be. I think it's too early to say from the lack of responses, though -- it's still midday Monday in California, where many of the core team live, so it hasn't been a full workday yet and someone who knows the answer may just not yet have seen the question.

Reply:
There’s no magic in `[zulip-rtd-git-cloning]`. It’s just a broken link. It was broken by #19993.

Reply:
(And the link in the readthedocs page was still working because RTD hadn't been updated, because of a separate issue. Details at [the chat thread](https://chat.zulip.org/#narrow/stream/19-documentation/topic/links.20to.20other.20documentation/near/1275536).)